pylearn2/utils/data_specs.py

```diff
@@ -265,7 +266,7 @@ def nest(self, flat):
                 assert self.n_unique_specs == 1
             return self._make_nested_space(flat, self.spec_mapping)
         else:
-            if isinstance(flat, tuple):
+            if isinstance(flat, Sized):
```

Comment:
I haven't read the code in detail, but I guess that changing all the `isinstance` checks is better? Just in case another method or the user decides to call these methods without converting, which would just re-introduce the bug.